src/Symfony/Component/Console/Attribute/Argument.php

```diff
@@ -22,20 +22,19 @@ class Argument
 {
     private const ALLOWED_TYPES = ['string', 'bool', 'int', 'float', 'array'];
 
+    private string|bool|int|float|array|null $default = null;
     private ?int $mode = null;
 
     /**
      * Represents a console command <argument> definition.
      *
-     * If unset, the `name` and `default` values will be inferred from the parameter definition.
+     * If unset, the `name` value will be inferred from the parameter definition.
      *
-     * @param string|bool|int|float|array|null                               $default         The default value (for InputArgument::OPTIONAL mode only)
-     * @param array|callable-string(CompletionInput):list<string|Suggestion> $suggestedValues The values used for input completion
+     * @param array<string|Suggestion>|callable-string(CompletionInput):list<string|Suggestion> $suggestedValues The values used for input completion
      */
     public function __construct(
         public string $name = '',
         public string $description = '',
-        public string|bool|int|float|array|null $default = null,
         public array|string $suggestedValues = [],
     ) {
         if (\is_string($suggestedValues) && !\is_callable($suggestedValues)) {
```

Comment:
why only callable-string BTW?

Reply:
Typing that as `callable` rather than `string` would indeed make more sense (and would automatically make the code compatible with static closures once PHP support them in attributes)

Reply:
Good idea! working on that...

Reply:
However, it's limited to callable-array so far